buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-							},
-							Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-						},
-					}); s != "" {
-						t.Fatalf("Unexpected:\n%s\n", s)
-					}
-				})
-			})
-
-			when("building fails", func() {
-				it("should error when the output buildpack plan is invalid", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkfile(t, "bad-key", filepath.Join(appDir, "build-plan-out-A-v1.toml"))
-					if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err == nil {
-						t.Fatal("Expected error.\n")
-					} else if !strings.Contains(err.Error(), "key") {
-						t.Fatalf("Incorrect error: %s\n", err)
-					}
-				})
-
-				it("should error when top level version and metadata version are both present and do not match", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkfile(t,
-						"[[entries]]\n"+
-							`name = "dep1"`+"\n"+
-							`version = "v2"`+"\n"+
-							"[entries.metadata]\n"+
-							`version = "v1"`+"\n",
-						filepath.Join(appDir, "build-plan-out-A-v1.toml"),
-					)
-					_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-					h.AssertNotNil(t, err)
-					expected := "top level version does not match metadata version"
-					h.AssertStringContains(t, err.Error(), expected)
-				})
-			})
-		})
-
-		when("buildpack api < 0.6", func() {
-			it.Before(func() {
-				bpTOML.API = "0.5"
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-			})
-
-			it("should include processes and set/override their default value to false", func() {
-				h.Mkfile(t,
-					`[[processes]]`+"\n"+
-						`type = "type-with-no-default"`+"\n"+
-						`command = "some-cmd"`+"\n"+
-						`[[processes]]`+"\n"+
-						`type = "type-with-default"`+"\n"+
-						`command = "other-cmd"`+"\n"+
-						`default = true`+"\n",
-					filepath.Join(appDir, "launch-A-v1.toml"),
-				)
-				br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-				if err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				if s := cmp.Diff(br, buildpack.BuildResult{
-					LaunchBOM:   nil,
-					Labels:      []buildpack.Label{},
-					MetRequires: nil,
-					Processes: []launch.Process{
-						{Type: "type-with-no-default", Command: "some-cmd", BuildpackID: "A", Default: false},
-						{Type: "type-with-default", Command: "other-cmd", BuildpackID: "A", Default: false},
-					},
-					Slices: []layers.Slice{},
-				}); s != "" {
-					t.Fatalf("Unexpected metadata:\n%s\n", s)
-				}
-				expected := "Warning: default processes aren't supported in this buildpack api version. Overriding the default value to false for the following processes: [type-with-default]"
-				assertLogEntry(t, logHandler, expected)
-			})
-
-			when("the launch, cache and build flags are in the types table", func() {
-				it("should warn", func() {
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L1152-L1166